cosmwasm/contracts/wormchain-ibc-receiver/src/contract.rs

```diff
@@ -0,0 +1,210 @@
+use anyhow::{ensure, Context};
+use cosmwasm_std::{entry_point, to_binary, Binary, Deps, Empty, Event, StdError, StdResult};
+use cosmwasm_std::{DepsMut, Env, MessageInfo, Order, Response};
+use cw2::{get_contract_version, set_contract_version};
+use semver::Version;
+use serde_wormhole::RawMessage;
+use wormhole::ibc_receiver::{Action, GovernancePacket};
+use wormhole::vaa::{Body, Header};
+use wormhole::Chain;
+use wormhole_bindings::WormholeQuery;
+
+use crate::error::ContractError;
+use crate::msg::{AllChainConnectionsResponse, ChainConnectionResponse, ExecuteMsg, QueryMsg};
+use crate::state::CHAIN_CONNECTIONS;
+
+// version info for migration info
+const CONTRACT_NAME: &str = "crates.io:wormchain-ibc-receiver";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    _msg: Empty,
+) -> Result<Response, anyhow::Error> {
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)
+        .context("failed to set contract version")?;
+
+    Ok(Response::new()
+        .add_attribute("action", "instantiate")
+        .add_attribute("owner", info.sender)
+        .add_attribute("version", CONTRACT_VERSION))
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn migrate(deps: DepsMut, _env: Env, _msg: Empty) -> Result<Response, anyhow::Error> {
```

Comment:
Where is this migrate logic from? It does not match how we perform migrates in the wormhole contract, and more specifically blocks upgrading from the wormhole contract. All of this versioning code appears new / different and causes this issue.

Reply:
I got it from here -- https://medium.com/cosmwasm/cosmwasm-for-ctos-ii-advanced-usage-ee04ce95d1d0

When we run the migrate from `wormhole` to `wormhole-ibc`, I was under the impression that only the migrate function in the `wormhole` contract would be called and this would pass.

I can remove these checks from both contracts though.